### PR TITLE
Add enterprise-extensions package

### PR DESCRIPTION
**Pull request by Hazboun6**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/main/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Hazboun6** [comment]
@AaronDJohnson and @paulthebaker please comment to let the nice folks at `conda-forge` know you are willing to be maintainers!

**Comment by AaronDJohnson** [comment]
I am willing to be a maintainer for this project 👍

**Comment by Hazboun6** [comment]
@AaronDJohnson any idea what went wrong here? Is there some incompatibility with `noarch` versus dictating a version of python?

**Comment by paulthebaker** [comment]
I'm willing to be a maintainer too!

**Comment by Hazboun6** [comment]
Okay, @conda-forge/help-python there was some serious linting shrapnel on this one, but I think we've figured out all of the subtleties. A few of the upstream packages for this are **not** available for windows, hence we needed to skip windows builds. I believe that precludes us from doing a `noarch` build, but if that is avoidable please let me know how I could change the recipe. Thanks!

**Comment by dopplershift** [comment]
This looks like pure Python. What's the reason for skipping on Windows?

**Comment by Hazboun6** [comment]
Hi @dopplershift , the previous comment basically tells the story. The build was failing on windows because a few of the upstream packages are not available for windows. You are correct though the package itself is pure python. If there is a way around skipping windows then I'm happy to do it.

**Comment by Hazboun6** [comment]
This was the error I was getting in the Azure builds:
```conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {"pint-pulsar[version='>=0.8.2']", "libstempo[version='>=2.4.0']", "scikit-sparse[version='>=0.4.5']", "healpy[version='>=1.14.0']"```

**Comment by dopplershift** [comment]
The way around is we say "Ok, it's failing due to missing dependencies. We'll just ignore the CI windows build since this is noarch." 😉 

So please go ahead and put this back to noarch Python since it drastically cuts down the builds.

**Comment by dopplershift** [comment]
This nice thing is that if the other dependencies start working on windows, this package is instantly installable without any changes.

**Comment by Hazboun6** [comment]
> The way around is we say "Ok, it's failing due to missing dependencies. We'll just ignore the CI windows build since this is noarch." 😉

Love it. 

@dopplershift Alright! I've made the recipe `noarch`!

**Comment by Hazboun6** [comment]
@dopplershift Just pinging you again to remind you that I think this is ready to be merged.

File: recipes/enterprise-extensions/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+    host:
+        - python >=3.6
+        - pip
+        - numpy >=1.16.3
+        - scipy >=1.2.0
+        - pytest
+
```
Comment (dopplershift) [inline recipes/enterprise-extensions/meta.yaml:24]: I don't think these are needed at build time (when running `setup.py`)?
```suggestion
```

File: recipes/enterprise-extensions/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        - scipy >=1.2.0
+        - ephem >=3.7.6.0
+        - healpy >=1.14.0
+        - scikit-sparse >=0.4.5
+        - pint-pulsar >=0.8.2
+        - libstempo >=2.4.0
+        - enterprise-pulsar >=3.1.0
+        - emcee
+        - ptmcmcsampler
+        - numdifftools
```
Comment (dopplershift) [inline recipes/enterprise-extensions/meta.yaml:37]: This is listed upstream in `requirements.txt`, but not in `install_requires` in `setup.py`, nor does it seem to be actually imported by the package?
```suggestion
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
I found a couple dependency things, but otherwise this looks in great shape.

**Review by dopplershift — APPROVED** [review]
Looks good. Ignoring windows failure due to missing deps.

**Pull request merged**